### DISCUSSION TITLE
Multithread/multicore CPU

### DISCUSSION THREAD

**Discussion by Yhontanac** [SWivid/F5-TTS#451] [Q&A]
Greetings,

I'm ussing a 4090 and a high tier .M2 for finetunning into my native languaje (debian. No other programs running, and so..).

I see the GPU at 100% load and 70º
But the CPU is ussing only one core (at 100%) and burring at >90º

For sure the problem is that I don't know a fck about how this works :sweat: 
But I was wondering if finetune_cli.py is (or could be) able to use multiple cores, so it could reduce CPU load (thus temperature).

Thanks for your time, and sorry for wasting it.

**Comment by sngraam** [comment]
may you use `f5-tts_infer-cli -c custom.toml ` that kind of staff is stff use self.device === 

```
self.device = device or (
            "cuda" if torch.cuda.is_available() else "mps" if torch.backends.mps.is_available() else "cpu"
        )
```

so you must use cuda is gpu GPUs to accelerate computing tasks chan "cpu" & id u have working support "mps" its be work and make sure you use ubuntu(recommended) if you not then change some settig in window may this is internal issue 

last way 
-- [ ice and hit cpu :) ]
